homeassistant/helpers/network.py

```diff
@@ -80,6 +83,7 @@ def get_url(
     require_current_request: bool = False,
     require_ssl: bool = False,
     require_standard_port: bool = False,
+    allow_hostname: bool = False,
```

Comment:
I think that we can apply this fix just to `async_process_play_media_url` and don't need to update `get_url`.

Are add-ons not capable of resolving the IP address of the host and connect to it?

Reply:
It depends on what the internal URL is. If the url they are given is `homeassistant.local` or the hostname of HA then yea we shouldn't have to do anything special. This is one of the benefits of my recent change to how MDNS works in `plugin-dns`. The system resolver gives us back all the answers it knows for a given domain so when asked about `homeassistant.local` it returns a response like this:
![image](https://user-images.githubusercontent.com/2037026/155265262-0753e8d9-68cc-4d49-a50b-096fa891d479.png)

I changed it so in the response it orders them like this:
1) Answer in the `172.30.32.0/23` subnet (if there is one)
2) Answer associated with the primary interface of the host
3) Any others in the order received from resolver

So basically - yea if the internal url is `homeassistant.local` it just works. But for any users that changed the internal URL to something else it won't. The only answer we'll have is whatever `eth0` (or whatever the primary interface is) sees. Which will resolve to a LAN IP.

This change makes it so no matter the internal URL is its possible to get back a URL which stays within the docker network if the integration plans to hand it to something managed by supervisor.

Reply:
If we know we want the internal url, we don't need to pass an "internal" url to `get_url` to force it to return us the internal url. 

So it could be: 

```python
if is_hassio:
    base_url = "homeassistant.local"
else:
    base_url = get_url(...)
```